backend/src/email/middlewares/email-transactional.middleware.ts

```diff
@@ -23,53 +26,120 @@ export const InitEmailTransactionalMiddleware = (
   authService: AuthService
 ): EmailTransactionalMiddleware => {
   const logger = loggerWithLabel(module)
+  interface ReqBody {
+    subject: string
+    body: string
+    from: string
+    recipient: string
+    reply_to: string
+    attachments?: { data: Buffer; name: string }[]
+  }
 
-  async function sendMessage(
+  async function saveMessage(
     req: Request,
     res: Response,
     next: NextFunction
   ): Promise<void> {
-    const ACTION = 'sendMessage'
+    const action = 'saveMessage'
+    logger.info({ message: 'Saving email', action })
+    const {
+      subject,
+      body,
+      from,
+      recipient,
+      reply_to: replyTo,
+      attachments,
+    }: ReqBody = req.body
 
-    try {
-      const {
+    const { fromName, fromAddress } = parseFromAddress(from)
+    const userEmail = (await authService.findUser(req.session?.user?.id))?.email
+    const emailMessageTx = await EmailMessageTx.create({
+      userEmail,
+      fromName,
+      fromAddress,
+      recipient,
+      params: {
         subject,
         body,
         from,
-        recipient,
-        reply_to: replyTo,
-        attachments,
-      } = req.body
+        replyTo,
+      },
+      hasAttachment: !!attachments,
+      attachmentS3Object: null,
+      status: TransactionalEmailMessageStatus.Unsent,
+      sentAt: null,
+      errorCode: null,
+      // not sure why unknown is needed to silence TS (yet other parts of the code base can just use `as Model` directly hmm)
+    } as unknown as EmailMessageTx)
+    if (!emailMessageTx) {
+      // TODO: log error, etc.
+      res.status(500).json({
+        message: 'Unable to create entry in email_message_tx',
+      })
+      throw new Error('Unable to create entry in email_message_tx')
+    }
+    // attach to body for subsequent middlewares
```

Comment:
modify comment to make clear:
1. save db calls (`userEmail`)
2. to tell whether the request is an API request (see `req.body.emailMessageTxId` used in `email.middleware.ts`